### PR TITLE
Feature/enable completions UX change in realtime

### PR DESCRIPTION
**Pull request by amircodota**
Support having the inline capabilities change affect immediately, without requiring a restart

### CODE REVIEW

File: src/capabilities/capabilities.ts [hunk 1]
```diff
@@ -49,11 +54,52 @@ export function fetchCapabilitiesOnFocus(): Promise<void> {
...
+
+export function onDidRefreshCapabilities(listener: () => void): Disposable {
+  return capabilitiesRefreshed.event(listener);
+}
+
+async function refreshCapabilities(): Promise<void> {
+  const capabilities = await getCapabilities();
+
+  enabledCapabilities = {};
+  capabilities?.enabled_features.forEach((feature) => {
```
Comment (dimacodota) [inline src/capabilities/capabilities.ts:73]:
```suggestion
  enabledCapabilities = capabilities?.enabled_features.map((feature) => ({[feature]: true})
```

Comment (amircodota) [reply src/capabilities/capabilities.ts:73]: This does not do the same thing

```
> let enabled_features = ['a', 'b', 'c'];
undefined
> 
> let enabledCapabilities = {};
undefined
> enabled_features.forEach((feature) => {
...     enabledCapabilities[feature] = true;
... });
undefined
> 
> enabledCapabilities
{ a: true, b: true, c: true }
> 
> enabled_features.map((feature) => ({[feature]: true}))
[ { a: true }, { b: true }, { c: true } ]
```

To use it properly, we can do something like

```
Object.assign({}, ...enabled_features.map((feature) => ({[feature]: true})))
```

Or maybe use `Array.reduce`

```
enabled_features.reduce((o, feature) => {
	o[feature] = true;
	return o;
}, {});
```

Both do not seem more readable to me than the original

File: src/capabilities/capabilities.ts [hunk 2]
```diff
@@ -49,11 +54,52 @@ export function fetchCapabilitiesOnFocus(): Promise<void> {
...
-
+  void refreshCapabilities().then(() => {
     resolve();
+    startRefreshLoop();
+  });
+}
+
+const capabilitiesRefreshed = new EventEmitter<void>();
+
+export function onDidRefreshCapabilities(listener: () => void): Disposable {
```
Comment (dimacodota) [inline src/capabilities/capabilities.ts:65]: it is bit strange that the `listener` does not get the refreshed capabilities as input, isn't it?

**Review by dimacodota — APPROVED** [review]
lgtm

**Pull request merged**